src/Microsoft.Health.Fhir.Core/Features/Operations/Import/ImportProcessingTask.cs

```diff
@@ -0,0 +1,274 @@
+                    baseUriString: _inputData.BaseUriString,
+                    correlationId: _inputData.TaskId,
+                    requestHeaders: new Dictionary<string, StringValues>(),
+                    responseHeaders: new Dictionary<string, StringValues>())
+            {
+                IsBackgroundTask = true,
+            };
+
+            _contextAccessor.RequestContext = fhirRequestContext;
+
+            CancellationToken cancellationToken = _cancellationTokenSource.Token;
+
+            long succeedImportCount = _importProgress.SucceedImportCount;
+            long failedImportCount = _importProgress.FailedImportCount;
+
+            ImportProcessingTaskResult result = new ImportProcessingTaskResult();
+            result.ResourceType = _inputData.ResourceType;
+
+            try
+            {
+                if (cancellationToken.IsCancellationRequested)
+                {
+                    throw new OperationCanceledException();
+                }
+
+                Func<long, long> sequenceIdGenerator = (index) => _inputData.BeginSequenceId + index;
+
+                // Clean imported resource after last checkpoint.
+                if (_importProgress.NeedCleanDataForRetry)
+                {
+                    await CleanDataAsync(cancellationToken);
+                }
+                else
+                {
+                    _importProgress.NeedCleanDataForRetry = true;
+                    await _contextUpdater.UpdateContextAsync(JsonConvert.SerializeObject(_importProgress), cancellationToken);
+                }
+
+                // Initialize error store
+                IImportErrorStore importErrorStore = await _importErrorStoreFactory.InitializeAsync(GetErrorFileName(), cancellationToken);
+                result.ErrorLogLocation = importErrorStore.ErrorFileLocation;
+
+                // Load and parse resource from bulk resource
+                (Channel<ImportResource> importResourceChannel, Task loadTask) = _importResourceLoader.LoadResources(_inputData.ResourceLocation, _importProgress.CurrentIndex, _inputData.ResourceType, sequenceIdGenerator, cancellationToken);
+
+                // Import to data store
+                (Channel<ImportProcessingProgress> progressChannel, Task importTask) = _resourceBulkImporter.Import(importResourceChannel, importErrorStore, cancellationToken);
+
+                // Update progress for checkpoints
+                await foreach (ImportProcessingProgress progress in progressChannel.Reader.ReadAllAsync())
+                {
+                    if (cancellationToken.IsCancellationRequested)
+                    {
+                        throw new OperationCanceledException("Import task is canceled by user.");
+                    }
+
+                    _importProgress.SucceedImportCount = progress.SucceedImportCount + succeedImportCount;
+                    _importProgress.FailedImportCount = progress.FailedImportCount + failedImportCount;
```

Comment:
This looks like it will not record all the success or failures if there are multiple progresses. The succeedImportCount and failedImportCount are not updated inside the foreach loop.

Reply:
for multiple progress, we have initial import count at beginning from context. So the result would resume from last checkpoint.
 long succeedImportCount = _importProgress.SucceedImportCount;
 long failedImportCount = _importProgress.FailedImportCount;